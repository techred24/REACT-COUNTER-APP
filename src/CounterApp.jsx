import { useState } from 'react';
import PropTypes from 'prop-types';
import React from 'react';
export function CounterApp({value}) {
    const [ counter, setCounter ] = useState(value);
    function handleApp(event, handler) {
        // setCounter(counter+1);
        setCounter((c) => c + 1);
    }
    const handleSubstract = () => setCounter(counter-1);
    const handleReset = () => setCounter(value);
    return (
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>

            <button onClick={handleApp}> +1 </button>
            <button onClick={handleSubstract}> -1 </button>
            <button aria-label='btn-reset' onClick={handleReset}> Reset </button>
        </>
    );
}
CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}