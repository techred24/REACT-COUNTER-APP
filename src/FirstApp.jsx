import PropTypes from 'prop-types';
const newMessage = {
    message: 'Mensage',
    name: 'Rafael'
};
const getMessage = () => 'Message from function';
export function FirstApp( props ) {
    console.log(props, 'Properties');
    console.log(props.customObj, 'Custom Object');
    const { name: nombre } = props.customObj;
    return  <>
                <code>{ JSON.stringify(newMessage) }</code>
                <div> {getMessage()} </div>
                <p>First App</p>
                <h2> {props.title} </h2>
                <h2> {props.subtitle} </h2>
                <p> {nombre} </p>
            </>
}
FirstApp.propTypes = {
    title: PropTypes.string.isRequired
}
FirstApp.defaultProps = {
    // title: 'Untitled like my fav band'
}