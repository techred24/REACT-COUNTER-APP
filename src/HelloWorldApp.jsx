import React from 'react';
import { FirstApp } from './FirstApp';
const myObj = {
    name: 'Flubber',
    age: 24
}
export const HelloWorldApp = () => {
  return (
    <>
        <div>HelloWorldApp mas</div>
        <FirstApp subtitle="Subtitulo mandado como prop" customObj={myObj} />
    </>
  )
}
