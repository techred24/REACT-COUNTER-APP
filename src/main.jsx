import React from 'react';
import ReactDOM from 'react-dom/client';
import { HelloWorldApp } from './HelloWorldApp';
import { CounterApp } from './CounterApp';
import { FirstApp } from './FirstApp';
import './styles.css';

ReactDOM.createRoot( document.getElementById('root') ).render(
    <React.StrictMode>
        <CounterApp value={24}/>
        {/* <FirstApp customObj={{name:'Flubber'}} /> */}
    </React.StrictMode>
);