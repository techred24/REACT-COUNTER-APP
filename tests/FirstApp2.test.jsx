import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";
import React from 'react';

describe('Pruebas en <FirstApp />', () => {
    const title = 'Hi there';

    test('Debe hacer match con el snapshot', () => {
        const { container } = render(<FirstApp title={ title } />);
        expect( container ).toMatchSnapshot();
    });
    test('Debe mostrar el mensaje "Flubberson"', () => {
        render( <FirstApp title={ title } /> );
        expect( screen.getByText(title) ).toBeTruthy();
    });
    test('Debe mostrar el titulo en un h1', () => {
        render( <FirstApp title={ title } />);
        expect( screen.getByRole('heading', { level: 1 }).innerHTML ).toContain( title );
    });
    test('Debe mostrar el subtitulo enviado por props', () => {
        const title = 'Hi, there';
        const subtitle = 'Hi, I am a subtitle';
        render( <FirstApp title={ title } subtitle={ subtitle } /> );
        expect( screen.getAllByText(subtitle).length ).toBe(2);
    });
});