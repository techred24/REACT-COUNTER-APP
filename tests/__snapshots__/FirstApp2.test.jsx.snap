// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <code>
    {"message":"Mensage","name":"Rafael"}
  </code>
  <div>
     
    Message from function
     
  </div>
  <p>
    First App
  </p>
  <h1
    data-testid="test-title"
  >
     
    Hi there
     
  </h1>
  <h2>
     
     
  </h2>
  <h2>
     
     
  </h2>
</div>
`;
